crd/src/lib.rs

```diff
@@ -95,19 +95,26 @@ pub struct SparkNode {
 #[derivative(Hash, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct SparkNodeSelector {
-    // common
+    // common options
     pub node_name: String,
     // we need to ignore instances from hashing -> if we scale up or down, the hash
     // for the selectors changes and the operator removes all nodes and rebuilds them
     #[derivative(Hash = "ignore", PartialEq = "ignore")]
     pub instances: usize,
     pub config: Option<Vec<ConfigOption>>,
     pub env: Option<Vec<ConfigOption>>,
-    // TODO: master -> use Option<T>
-    // worker -> use Option<T>
+
+    // master options
+    pub master_port: Option<usize>,
```

Comment:
This is a bit ... mehh.. to have master & worker stuff in the same selector (I know we talked about it in the past). But I'm fine with leaving it for now.

Reply:
I know it looks somewhat weird but has some benefits. Less code, better to work with. Restriction are coming from CRD here. So you cannot assign the master_port for workers etc. 
For now it works, but i think we probably have to split them up at some point too.